enhancements/proposals/1978-scylladbdatacenter-api/README.md

```diff
@@ -0,0 +1,711 @@
+  annotation `internal.scylla-operator.scylladb.com/enable-host-networking` to `true` on `v1alpha1.ScyllaDBDatacenter`.
+  The controller migrating from `v1.ScyllaCluster` to `v1alpha1.ScyllaDBDatacenter` will set this annotation when required.
+- `sysctls` doesn't belong to ScyllaDB cluster nor datacenter, these are configuration options for nodes. 
+  A better place to have them is NodeConfig. Until it's available in NodeConfig, we will support setting them 
+  via `internal.scylla-operator.scylladb.com/set-sysctls` annotation, in a JSON encoded array of `"key": "value"` format. [#749](https://github.com/scylladb/scylla-operator/issues/749)
+
+### Helm chart
+
+The existing `scylla` helm chart will use the existing `v1.ScyllaCluster`. It has several differences between what's in CRD
+and what's in the chart templates, hence it cannot be simply mapped one-to-one.
+A new helm chart called `scylladb-datacenter` will simply copy `spec` from `values` to a new CRD solving this issue and making it
+easily extensible, while keeping the schema of CRD and helm schema in sync.
+
+### Scylla Manager integration
+
+Scylla Manager tasks do not fit into a single datacenter, as Manager tasks work on the entire cluster.
+These are not yet transferred from the spec and status of the old API until a better alternative for these is available.
+The existing controller reconciling these will continue using `v1.ScyllaCluster`, users wanting to use these,
+won't be able to transition to the new API until an extension is available.
+Extending the manager integration may be addressed in a future proposal that rethinks the overall approach in more detail.
+
+### Topology placement
+
+Currently, to provide placement rules for a rack, users have to provide a very lengthy `nodeAffinity`.
+Even the simplest rules require providing 10'ish lines with 4'ish indentation levels.
+This makes the entire resource huge and UX isn't great.
+To improve it, the new field `topologyLabelSelector` added on the datacenter and rack level,
+provides a label selector which will be translated into placement nodeAffinity rules selecting nodes matching these labels.
+Label selectors from the datacenter and rack level will be merged into the final node affinity. 
+In case of a collision, rack ones take precedence.
+
+Given following spec:
+```yaml
+spec:
+  rackTemplate:
+    topologyLabelSelector:
+      scylla.scylladb.com/node-type: scylla
+  racks:
+  - name: a
+    topologyLabelSelector:
+      topology.kubernetes.io/zone: us-east1a
+```
+
+Results in rack `a` having `nodeAffinity`:
+```yaml
+nodeAffinity:
+  requiredDuringSchedulingIgnoredDuringExecution:
+    nodeSelectorTerms:
+      - matchExpressions:
+        - key: scylla.scylladb.com/node-type
+          operator: In
+          values:
+            - scylla
+        - key: topology.kubernetes.io/zone
+          operator: In
+          values:
+          - us-east1a
```

Comment:
nit: remove the extra indent for yaml arrays for consistency

Reply:
done

Reply:
the whole thing under `nodeSelectorTerms` as well

Reply:
done